### PR TITLE
Fix #4327 radia extruded shapes from points file

### LINKED ISSUE #4327: Add semi-arbitrary extruded shapes
Expand the extruded shapes to include:

- <s>points drawn (or added) by hand</s>
- points read from a file
- <s>points generated by pre-defined functions (quadratics etc.)</s>

(save the crossed-off items for future issues)

This will be especially useful for building pole tips.

- [x] Import from file
- [ ] Match shape in freehand editor
- [ ] Add canvas with draggable points
- [ ] Simplify UI to avoid nested dialogs

### PR DESCRIPTION
**Pull request by mkeilman**
Notes:
- I scaled this back to handle only points defined in a file. Once imported the points can be viewed but not changed
- Also deferred is an in-editor plot, though some of the plumbing is in place
- The points are interpreted as relative to the shape center, so they remain constant as the user moves the center around. The order is in order of a permutation of the axes; that is, if the extrusion axis is Z, then the first coordinate of the point is X and the second is Y. Similarly X -> (Y, Z) and Y -> (Z, X)
- To try it out, make a new freehand simulation, switch to the designer tab, and drag an extruded object from the toolbar to the workspace. Click the shape button on the editor that pops up, then the Point Editor tab, then choose a file

### CODE REVIEW

File: sirepo/package_data/static/js/radia.js [hunk 1]
```diff
@@ -30,24 +30,33 @@ SIREPO.app.config(function() {
...
         </div>
         <div data-ng-switch-when="PtsFile" data-ng-class="fieldClass">
-          '<input id="radia-pts-file-import" type="file" data-file-model="model[field]" accept=".dat,.txt"/>
+          <input id="radia-pts-file-import" type="file" data-file-model="model[field]" accept=".dat,.txt,.csv"/>
+        </div>
+        <div data-ng-switch-when="Points" data-ng-class="fieldClass">
+          <label class="control-label col-sm-5" style="text-align: center">{{ model.widthAxis }}</label> <label class="control-label col-sm-5"  style="text-align: center">{{ model.heightAxis }}</label>
+          <div class="col-sm-12" style="height: 200px; overflow-y: scroll; overflow-x: hidden;">
+              <div data-ng-repeat="p in model[field]">
+                <input class="form-control sr-number-list" data-string-to-number="float" data-ng-model="p[0]" data-ng-disabled="model.pointsFile" style="text-align: right;" required />
```
Comment (gurhar1133) [inline sirepo/package_data/static/js/radia.js:53]: does p have more than 2 elems? Maybe could be one input
```html
<input ... data-ng-repeat="e in p" data-ng-model="e" .../>

File: sirepo/package_data/static/js/radia.js [hunk 2]
```diff
@@ -2040,6 +2075,25 @@ SIREPO.app.directive('kickMapReport', function(appState, panelState, plotting, r
...
+            modelName: '=',
+            model: '=',
+        },
+        template: `
+            <div>
+                <div data-ng-repeat=""></div>
+            </div>
+        `,
+        controller: function($scope, $element) {
+            //<div data-report-content="parameter" data-model-key="extrudedPoly" data-report-id=""></div>
```
Comment (gurhar1133) [inline sirepo/package_data/static/js/radia.js:2092]: Don't completely understand the commented out html

File: sirepo/package_data/static/js/radia.js [hunk 3]
```diff
@@ -3699,16 +3750,46 @@ SIREPO.viewLogic('objectShapeView', function(appState, panelState, radiaService,
...
         modelType = appState.models.geomObject.type;
         $scope.modelData = appState.models[$scope.modelName];
-        radiaService.updateModelAndSuperClasses(modelType, $scope.modelData);
-        updateObjectEditor();
+        editedModels = radiaService.updateModelAndSuperClasses(modelType, $scope.modelData);
+        updateShapeEditor();
     };
 
+    $scope.$on('geomObject.changed', () => {
+        //appState.saveQuietly($scope.modelName);
```
Comment (gurhar1133) [inline sirepo/package_data/static/js/radia.js:3764]: Why the empty callback?

File: sirepo/template/radia.py [hunk 4]
```diff
@@ -251,6 +261,23 @@ def post_execution_processing(
     return template_common.parse_mpi_log(run_dir)
 
 
+def stateful_compute_build_shape_points(data):
+    import csv
+
+    pts = []
+    with open(
```
Comment (gurhar1133) [inline sirepo/template/radia.py:268]: why not pkio.read_text?

**Review by gurhar1133 — APPROVED** [review]
Looks good. Left a couple comments/questions, will approve anyways so up to you if you want to change

File: sirepo/package_data/static/js/radia.js (hunk 1 above)
Comment (mkeilman) [reply sirepo/package_data/static/js/radia.js]: [a47c51d](https://github.com/radiasoft/sirepo/pull/4686/commits/a47c51dfad824e6f5f04f57a903dcafc32d704ab)

Had to include `track by $index` to handle duplicate values

File: sirepo/package_data/static/js/radia.js (hunk 3 above)
Comment (mkeilman) [reply sirepo/package_data/static/js/radia.js]: [4568a69](https://github.com/radiasoft/sirepo/pull/4686/commits/4568a69bab62665ae5c3906190faf5e9c6c98068)

Left over from previous work

File: sirepo/template/radia.py (hunk 4 above)
Comment (mkeilman) [reply sirepo/template/radia.py:268]: `csv` puts things into rows and columns for you. I prefer it to `pkio.read_text()` for this specific purpose

File: sirepo/package_data/static/js/radia.js (hunk 2 above)
Comment (mkeilman) [reply sirepo/package_data/static/js/radia.js]: [52f0150](https://github.com/radiasoft/sirepo/pull/4686/commits/52f015084417fcf016db76b1785b29e7616c95b7)

It was to have been a mini-plot of the shape, but I reduced the scope of the issue. No need to keep it

**Pull request merged**